File: 백준/Silver/1193. 분수찾기/분수찾기.py
N = int(input())
x=1
y=1
roll = 0
z=1
while 1:  
    if z == N:
        print("%d/%d"%(x,y))
        break

    if x-1 == 0:
        y+=1
        roll = 1
        z+=1
    elif y-1 == 0:
        x+=1
        roll = 0
        z+=1
    
    if z == N:
        print("%d/%d"%(x,y))
        break

    if roll == 1:
        x+=1
        y-=1
        z+=1
    elif roll == 0:
        x-=1
        y+=1
        z+=1

File: 백준/Silver/1193. 분수찾기/분수찾기.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.PerformanceData;
using System.Runtime.InteropServices;

namespace baekjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            int num = 0, num1 = 0,count=0;
            int leftnum , rightnum;
            int input = int.Parse(Console.ReadLine());
            int numb = 0;
            while (true)
            {

                num += 1;
                if (num % 2 == 0)
                {
                    leftnum = 1;
                    rightnum = num;
                }
                else
                {
                    leftnum = num;
                    rightnum = 1;
                }


                for (int i =1; i <= num;i++)
                {   count += 1;
                    if (count == input)
                    {
                        numb = 1;
                        break;
                    }
                    if (num % 2 == 0)
                    {
                        leftnum += 1;
                        rightnum -= 1;
                    }
                    else
                    {
                        leftnum -= 1;
                        rightnum += 1;
                    }
                }

                if (numb == 1)
                {
                    break;
                }
            }
            Console.WriteLine(leftnum + "/" + rightnum);
        }
    }
}        